Add documentation about creating initial user (#7433)
inmanta/inmanta-core: docs/administrators/auth.rst

.. _auth-setup:

Setting up SSL and authentication
=================================

This guide explains how to enable ssl and setup authentication.

SSL
---

> This section explain how to setup SSL. SSL is not strictly required for authentication but highly recommended.

Reviewer: ```suggestion
This section explain how to setup SSL. SSL is not strictly required for authentication but it is highly recommended.
```

> Inmanta uses bearer tokens for authorizing users and services. These tokens should be kept private and are visible

Reviewer: ```suggestion
Inmanta uses bearer tokens to authorize users and services. These tokens should be kept private and are visible
```

    issuer=https://localhost:8888/
    audience=https://localhost:8888/

To generate a secure symmetric key and encode it correctly use the following command:

.. code-block:: sh

    openssl rand 32 | python3 -c "import sys; import base64; print(base64.urlsafe_b64encode(sys.stdin.buffer.read()).decode().rstrip('='));"

.. _auth-int:

Built-in authentication provider
--------------------------------

The Inmanta server has a built-in authentication provider. This provider stores the authentication and authorization
information into the PostgreSQL database. As such, there is no need to rely on a 3rd party auth broker. The sections

> below describe how to enable the built-in authentication provider and how to create the initial admin user.

Reviewer: ```suggestion
below describe how to enable the built-in authentication provider and how to create the initial admin user.
Additional users can then be created via the API or through the web console.
```
